### DISCUSSION TITLE
Portfolio optimization with benchmark and turnover  constraints for asset classes

### DISCUSSION THREAD

**Discussion by FeCol150** [dcajasn/Riskfolio-Lib#187] [Q&A]
Hi,
First of all. Thank you so much again for the development in this library. Anytime I have the chance to implement something using it I got impressed of the great job you have done.

This time, I am working on a portfolio optimization. The idea is to have some assets grouped by asset classes the same than its benchmark. So I would like to optimize it according to the weight of the benchmark by asset class to constraint the turnover is it possible to do it wih the current status of the library or it is just done to do by asset?

Regards,

**Comment by dcajasn** [comment]
Hi @FeCol150,

No that's not possible now, the turnover constraint now is by asset level. One trick you can do is put a constraint on minimum level and maximum level per class based on benchmark and then use a tracking error constraint.

Best regards,
Dany
